yaml-tests/src/test/java/MultiServerConnectionFactoryTest.java

```diff
@@ -42,61 +43,119 @@ void testDefaultPolicy(int initialConnection) throws SQLException {
+        assertStatement(statement, "Primary");
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, "Primary");
     }
 
     @ParameterizedTest
     @CsvSource({"0", "1"})
     void testAlternatePolicy(int initialConnection) throws SQLException {
+        final String[] path = new String[] { "Primary", "Alternate" };
+
         MultiServerConnectionFactory classUnderTest = new MultiServerConnectionFactory(
                 MultiServerConnectionFactory.ConnectionSelectionPolicy.ALTERNATE,
                 initialConnection,
-                dummyConnectionFactory(),
-                List.of(dummyConnectionFactory()));
+                dummyConnectionFactory("Primary"),
+                List.of(dummyConnectionFactory("Alternate")));
         assertEquals(initialConnection, classUnderTest.getCurrentConnectionSelector());
-        MultiServerConnectionFactory.MultiServerRelationalConnection connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals(initialConnection % 2, connection.getCurrentConnectionSelector());
+
+        // First run:
+        // - Factory current connection: initial connection
+        // - connection current connection: initial connection
+        // - statement: initial connection
+        MultiServerConnectionFactory.MultiServerRelationalConnection connection =
+                (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
         assertEquals((initialConnection + 1) % 2, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, initialConnection, path[initialConnection]);
+        RelationalPreparedStatement statement = connection.prepareStatement("SQL");
+        assertStatement(statement, path[initialConnection]);
+        // next statement
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, path[(initialConnection + 1) % 2]);
+
+        // Second run:
+        // - Factory current connection: alternate connection
+        // - connection current connection: alternate connection
+        // - statement: initial connection
         connection = (MultiServerConnectionFactory.MultiServerRelationalConnection)classUnderTest.getNewConnection(URI.create("Blah"));
-        assertEquals((initialConnection + 1) % 2, connection.getCurrentConnectionSelector());
         assertEquals(initialConnection % 2, classUnderTest.getCurrentConnectionSelector());
+        assertConnection(connection, (initialConnection + 1) % 2, path[(initialConnection + 1) % 2]);
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, path[(initialConnection + 1) % 2]);
+        // next statement
+        statement = connection.prepareStatement("SQL");
+        assertStatement(statement, path[initialConnection]);
```

Comment:
At least one of these should have 3 or 1 calls to `prepareStatement`

Reply:
Good point, done.